File: src-tauri/src/models/vault_item.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct VaultItem {
    pub name: String,
    pub username: String,
    pub password: String,
    pub urls: Vec<String>,
    pub notes: String,

    pub is_favourite: bool,
    pub category: String,
}

File: src/types/vaultItem.ts
import { VaultItemCategory } from "../enums/vaultItemCategory";

interface VaultItem {
    name: string;
    username: string;
    password: string;
    urls: string[];
    notes: string;

    is_favourite: boolean;
    category: VaultItemCategory;
}

export type { VaultItem };